### PR TITLE
Feat(invitations): Generate an invitation to workspace, and make the link user-workspace

### PR DESCRIPTION
**Pull request by ImMPrada**
This PR enables an admin to generate an invitation link to add a user to a workspace.

Rules:
- only an admin can add a new member to a workspace
- the generated link can be used once for linking a member to the workspace
- if the invited user declines the invitation, the link is still valid

the link is generated and must be copied in order to share it, for example, https://ed86dd711635.ngrok.app/invitations/711160c707bbbf11fbcddafec99a9356b96c01fe
image

By visiting the invitation link, an invited user can enjoy the workspace
image

An already member user, can't enjoy again the workspace, and the invitation link keeps active
image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,3 +6,11 @@ run seeds to create the roles:
 ```bash
 rails db:seed
 ```
+
+## Authorization
+
+Is handled by pundit gem, taking into account that the chat has three roles: `member` `admin` `owner`
```
Comment (mapra99) [inline README.md]:
```suggestion
It's handled by pundit gem, taking into account that the chat has three roles: `member` `admin` `owner`
```

```suggestion
Authorization is handled by pundit gem, taking into account that the chat has three roles: `member` `admin` `owner`
```

File: app/models/invitation.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Invitation < ApplicationRecord
+  belongs_to :workspace
+  belongs_to :receiver, class_name: 'User', optional: true
+
+  validates :token, presence: true, uniqueness: true
+
+  def active?
+    active
+  end
```
Comment (mapra99) [inline app/models/invitation.rb]: if `active` is a boolean column, ActiveRecord will already provide this `active?` method for you, so you don't need to set it manually.

File: app/policies/workspace_policy.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  def create_invitation?
+    user.admin?(workspace)
+  end
+
+  class Scope < Scope
+    # NOTE: Be explicit about which records you allow access to!
+    # def resolve
+    #   scope.all
+    # end
+  end
```
Comment (mapra99) [inline app/policies/workspace_policy.rb:19]: don't you need to define a scope here?

**Review by mapra99 — APPROVED** [review]
looks good! I left some small comments

File: app/models/invitation.rb (hunk 2 above)
Comment (ImMPrada) [reply app/models/invitation.rb]: dah! Done!

**Pull request merged**